maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -4215,4 +4213,14 @@ private Classpath setCachedClasspath( @Nonnull String key, @Nonnull Set<Artifact
             return classpath;
         }
     }
+
+    /**
+     * No tests run failures.
+     */
+    enum NoTestsFailureReason
```

Comment:
Maybe it is better to get away from all this "NOs" I think "TestRunExpectation" or something similar might be better here ...

Reply:
It could be named `TestRunExpectation` in terms of positive outcome from the plugin execution.
Now the enum is used only in the method `getEffectiveFailIfNoTests()` and this method evaluates two config parameter if no tests were executed. So it is negative result of the plugin, see this:
```
            DefaultScanResult scan = scanForTestClasses();
            if ( !hasSuiteXmlFiles() && scan.isEmpty() )
            {
                switch ( getEffectiveFailIfNoTests() )
                {
                    case NO_TESTS_RUN:
                        throw new MojoFailureException(
                            "No tests were executed!  (Set -DfailIfNoTests=false to ignore this error.)" );
                    case NO_SPECIFIED_TESTS_RUN:
                        throw new MojoFailureException( "No tests matching pattern \""
                            + getSpecificTests().toString()
                            + "\" were executed! (Set "
                            + "-D" + getPluginName() + ".failIfNoSpecifiedTests=false to ignore this error.)" );
```

Reply:
This looks good to me and is much more clearer.

Reply:
I am also to 100% convinced about the names. It must depict the real meaning of usage in the code, as the one above. Feel free to propose better ones.

Reply:
Pls check the names of enum now, see the [commit](https://github.com/apache/maven-surefire/pull/350/commits/3e8c22489648e06a520771c5b857088bad0e565b).